pkg/transport/http/transport.go

```diff
@@ -0,0 +1,344 @@
+	}
+
+	response, err := t.handleIncomingMessage(requestData)
+	if err != nil {
+		t.logger.Error("Failed to process request", slog.String("error", err.Error()))
+		http.Error(w, "failed to process request", http.StatusInternalServerError)
+		return
+	}
+
+	t.setupHeaders(w, response)
+	t.setupContent(w, response)
+}
+
+func (t *Transport) HandleGeneralRequest(req *http.Request, w http.ResponseWriter, _ transport.OnCertificatesReceivedFunc) error {
+	requestID := req.Header.Get(requestIDHeader)
+	if requestID == "" {
+		t.logger.Error("Missing request ID, checking if unauthenticated requests are allowed")
+
+		if t.allowUnauthenticated {
+			return nil
+		}
+
+		return fmt.Errorf("missing request ID")
+	}
+
+	t.logger.Debug("Received general request", slog.String("requestID", requestID))
+
+	requestData, err := t.buildAuthMessageFromRequest(req)
+	if err != nil {
+		t.logger.Error("Failed to build request data", slog.String("error", err.Error()))
+		return err
+	}
+
+	_, err = t.handleIncomingMessage(requestData)
+	if err != nil {
+		t.logger.Error("Failed to process request", slog.String("error", err.Error()))
+		return fmt.Errorf("failed to process request")
+	}
+
+	t.setupHeaders(w, requestData)
+
+	return nil
+}
+
+func (t *Transport) handleIncomingMessage(msg *transport.AuthMessage) (*transport.AuthMessage, error) {
+	if msg.Version != transport.AuthVersion {
+		return nil, fmt.Errorf("unsupported version")
+	}
+
+	switch msg.MessageType {
+	case transport.InitialRequest:
+		return t.handleInitialRequest(msg)
+	case transport.General:
+		return msg, t.handleGeneralRequest(msg)
+	default:
+		return nil, fmt.Errorf("unsupported message type")
+	}
+}
+
+func (t *Transport) handleInitialRequest(msg *transport.AuthMessage) (*transport.AuthMessage, error) {
+	if msg.IdentityKey == "" && msg.InitialNonce == "" {
+		return nil, fmt.Errorf("missing required fields in initial request")
+	}
+
+	sessionNonce, err := t.wallet.CreateNonce(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create session nonce")
+	}
+
+	session := sessionmanager.PeerSession{
+		IsAuthenticated: true,
+		SessionNonce:    &sessionNonce,
+		PeerNonce:       &msg.InitialNonce,
+		PeerIdentityKey: &msg.IdentityKey,
+		LastUpdate:      time.Now(),
+	}
+	t.sessionManager.AddSession(session)
+
+	signature, err := createAuthSignature(t.wallet, msg.InitialNonce, sessionNonce, msg.IdentityKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create signature")
+	}
+
+	identityKey, err := t.wallet.GetPublicKey(context.Background(), wallet.GetPublicKeyOptions{IdentityKey: true})
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve identity key")
+	}
+
+	initialResponseMessage := transport.AuthMessage{
+		Version:      transport.AuthVersion,
+		MessageType:  "initialResponse",
+		IdentityKey:  identityKey,
+		InitialNonce: sessionNonce,
+		YourNonce:    &msg.InitialNonce,
+		//Certificates:          certificatesToInclude,
```

Comment:
Could you please leave TODO here with task number from github (please add it, even a placeholder if it doesn't exists).